File: frontend/src/models/Visit.ts
import {SpecialistProfileInfo} from "./Specialist";
import {Client} from "./Client";
import {SpecialistService} from "./SpecialistService";
import {ClientAddressExtended} from "./ClientAddress";

export interface Visit {
    id: number
    start_date: Date
    end_date: Date
    price: number
    description: string
    status: string
    client_address: ClientAddressExtended
}

export interface VisitCalendar {
    info: Visit
    specialist: SpecialistProfileInfo
    client: Client
    service: SpecialistService
}

export interface VisitRequest {
    start_date: Date
    end_date: Date
    description: string
    client_id: number
    client_address_id: number
    specialist_id: number
    specialist_service_id: number
}

export interface VisitBasicInfo {
    id: number
    service: string
}

File: backend/internal/models/Visit.go
package models

import "time"

type Visit struct {
	Id            int           `json:"id"`
	StartDate     time.Time     `json:"start_date"`
	EndDate       time.Time     `json:"end_date"`
	Price         int           `json:"price"`
	Description   string        `json:"description"`
	Status        string        `json:"status"`
	ClientAddress ClientAddress `json:"client_address"`
}

type VisitCalendar struct {
	Info       Visit                 `json:"info"`
	Specialist SpecialistProfileInfo `json:"specialist"`
	Client     Client                `json:"client"`
	Service    SpecialistService     `json:"service"`
}

type VisitRequest struct {
	StartDate           time.Time `json:"start_date"`
	EndDate             time.Time `json:"end_date"`
	Description         string    `json:"description"`
	ClientId            int       `json:"client_id"`
	ClientAddressId     int       `json:"client_address_id"`
	SpecialistId        int       `json:"specialist_id"`
	SpecialistServiceId int       `json:"specialist_service_id"`
}
